### DISCUSSION TITLE
How to set up a dynamic resource

### DISCUSSION THREAD

**Discussion by fabiopicchi** [dagster-io/dagster#20360] [Q&A]
I have defined a job that requires a myriad of parameters so, to simplify things, I created a config_mapping function. The ops in this job, however, all depend on a dataset ID (internal entity in our backend). I figured I could create this dataset ID inside the config_mapping function but, to my surprise, it is called again and again for every operation. 

My trick was to run the API call that creates a new dataset based on the job config inside the `config_mapping` function and use the generated dataset ID to define a resource for the job that can be shared across `ops`. This strategy, however, falls apart if `config_mapping` is invoked for every op. Is there another way to achieve the behavior I am seeking? I considered chaining two jobs: one to create the dataset and another to run the actual load. That, however, burdens the user with more parameters to set and jobs to monitor/run. I would really like to run the whole thing at once.

**Comment by alangenfeld** [comment]
Is your `config_mapping`  attached only on the `@job`? It may be unintentional that the function is running repeatedly. Maybe a small example that demonstrates would be useful.

Another approach could be to cache/memoize the dataset id generation against some value that does stay constant across the repeated invocations. Hard to if this idea has much merit without more context.
